integration/auction_test.sh

```diff
@@ -21,37 +24,103 @@ CC_VERS=0
 num_rounds=3
 num_clients=10
 
+# Try function which returns the response string
+try_r() {
+    echo "$@" 
+    export RESPONSE=""
+    "$@" 2>&1 | tee -a /tmp/response.txt || die "test failed: $*"
```

Comment:
hmm, this fails only if tee has an error, not if "$@" fails!

In bash you can access the exit codes via ${PIPESTATUS[0]}, ${PIPESTATUS[1]}, ${PIPESTATUS[2]}, ...

In fact, do you even need a pipe and a temp-file here? why not directly do `result=$("$@" 2>&1) and then test exit-code via $?

Reply:
I figured there was a better way to do this, just couldn't get the syntax correct.  This seems to work, however
```
try_r() {
    echo "$@" 
    result=$("$@" 2>&1) || die "test failed: $*"
    echo $result
    export RESPONSE=$(echo $result | awk -F "\"" '{print $5}' | awk -F "\\" '{print $1}' | base64 -d)
    say $RESPONSE
}
``` It goes to the die if the invoke fails.